nixos/modules/system/boot/loader/generations-dir/generations-dir-builder.sh

```diff
@@ -77,12 +78,13 @@ addEntry() {
       cp "$initrd" /boot/nixos-initrd
       cp "$(readlink -f "$path/init")" /boot/nixos-init
 
-      mkdir -p /boot/default
-      # ln -sfT: overrides target even if it exists.
-      ln -sfT $(readlink -f $path) /boot/default/system
-      ln -sfT $(readlink -f $path/init) /boot/default/init
-      ln -sfT $extraLnArgs $initrd /boot/default/initrd
-      ln -sfT $extraLnArgs $kernel /boot/default/kernel
+      # Clean up the /boot/default directory created by earlier
+      # versions of generations-dir.
+      if test -d $defaultdir -a ! -L $defaultdir; then
```

Comment:
This feels very dangerous. Maybe at least log a warning?

Reply:
The delete is required. If `$defaultdir` is not deleted, then the following `ln` would error out with `ln: default: cannot overwrite directory`.

Further up, there is an unconditional `rm -Rf /boot/system*`. The updates are already destructive and not atomic. Improvements in those areas would be nice, but is it a blocker?

Is the concern that users are storing files in `/boot/default` other than those created by `generationsDir`? If so, then what kind of warning would help? At this point in the script, we either finish setting up `/boot` or the user is left with an unbootable system.

Reply:
It's always hard to tell what is in scope of the bootloader script. IMO if there's an existing directory that contains anything but the files we're about to put there, we should just abort.